bug_1292234.patch

```diff
@@ -128,74 +128,80 @@ diff --git a/devtools/server/actors/storage.js b/devtools/server/actors/storage.
 +    }
 +  },
 +
-+  async populateStoreMap(host, storagePrincipal, storeMap) {
-+    const db = await ExtensionStorageIDB.open(storagePrincipal);
-+    this.dbConnectionForHost.set(host, db);
-+    const data = await db.get();
++  async getStoragePrincipal(extension) {
++    if (extension.getSharedData("storageIDBBackend")) {
++      return extension.getSharedData("storageIDBPrincipal");
++    }
++    // This extension either isn't using the IDB backend (false) or
++    // it hasn't migrated to the IDB backend yet (null)
 +
-+    for (const [key, value] of Object.entries(data)) {
-+      storeMap.set(key, value);
++    // Loop for web extension context related to the current window
++    let currentContext;
++    for (const view of extension.views) {
++      currentContext = view;
++    }
++    if (!currentContext) {
++      // We didn't find the extension context for the current target; give up.
++      return null;
```

Comment:
When would we not be able to find a `currentContext`? This method only executes when the storage inspector for the add-on toolbox is opened.

Should we throw an error here? Or a warning?

Reply:
> When would we not be able to find a currentContext? This method only executes when the storage inspector for the add-on toolbox is opened.

The scenario that I had in mind is the following one:
if the current target window is not an extension page, we may not be able to find an extension context related to it.

But I didn't actually verified if and when the above scenario can be actually triggered, we should look into building a test case that triggers it as part of that test file added in the same patch.

> Should we throw an error here? Or a warning?

I'm not sure if this is actually an error scenario, at a first glance sounds like it may be just a transient state (like the storage panel may just not listing the extension storage until an extension page is selected), and so throwing may not be the right reaction to it.

Reply:
Thanks for the clarification.

> The scenario that I had in mind is the following one:
if the current target window is not an extension page, we may not be able to find an extension context related to it.

**TLDR**: After digging into this a bit, I think this problem manifests further upstream of `currentContext`, since `currentContext` is defined in `populateStoresForHost`, and this method never gets called when the target window is not an extension page.

How do I know `populateStoresForHost` never gets called?

_Case 1:_
Installing my [test extension](https://github.com/biancadanforth/webextensions-examples/pull/3) with no background page and no other extension pages yet opened (therefore, no extension window to target), I opened the Storage Inspector via `about:debugging`. In this case, we never actually call `populateStoresForHost`, as its caller, `preListStores`, has a `null` value for `host` for the target window. Since no extensions scripts have yet run in this case, we do not have any items in extension storage, and the Storage Inspector correctly shows nothing in extension storage. This is not in itself a bug.

_Case 2:_
However, this is similarly the case even after I have opened a non-background extension page (in this case a browserAction popup that adds a couple storage items) in that same test extension and then open the Storage Inspector. In this situation, that popup script has added two items to extension local storage, and the storage inspector is incorrectly reporting no items. This is a bug!

All this is to say, while I think we have the right idea in a test case for when the current target window is not an extension page, handling that may need to happen further upstream in `preListStores`. I have filed Issue #9 and referenced it in the testing meta issue.